modules/sitemaps/sitemap-builder.php

```diff
@@ -0,0 +1,1458 @@
+
+		$jetpack_version = JETPACK__VERSION;
+
+		$namespaces = Jetpack_Sitemap_Buffer::array_to_xml_attr_string(
+			/**
+			 * Filter the XML namespaces included in video sitemaps.
+			 *
+			 * @module sitemaps
+			 *
+			 * @since 4.7.0
+			 *
+			 * @param array $namespaces Associative array with namespaces and namespace URIs.
+			 */
+			apply_filters(
+				'jetpack_sitemap_video_ns',
+				array(
+					'xmlns:xsi'          => 'http://www.w3.org/2001/XMLSchema-instance',
+					'xsi:schemaLocation' => 'http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd',
+					'xmlns'              => 'http://www.sitemaps.org/schemas/sitemap/0.9',
+					'xmlns:video'        => 'http://www.google.com/schemas/sitemap-video/1.1',
+				)
+			),
+			"\n " // This argument pretty prints the namespaces one per line.
+		);
+
+		$buffer = new Jetpack_Sitemap_Buffer(
+			JP_SITEMAP_MAX_ITEMS,
+			JP_SITEMAP_MAX_BYTES,
+			<<<HEADER
+<?xml version='1.0' encoding='UTF-8'?>
+<!-- generator='jetpack-{$jetpack_version}' -->
+<?xml-stylesheet type='text/xsl' href='{$video_sitemap_xsl_url}'?>
+<urlset{$namespaces}>\n
+HEADER
+			,
+			<<<FOOTER
+</urlset>\n
+FOOTER
+			,
+			/* epoch */
+			'1970-01-01 00:00:00'
+		);
+
+		// Add as many items to the buffer as possible.
+		while ( false === $buffer->is_full() ) {
+			$posts = $this->librarian->query_videos_after_id(
+				$last_post_id, JP_SITEMAP_BATCH_SIZE
+			);
+
+			if ( null == $posts ) { // WPCS: loose comparison ok.
+				$any_posts_left = false;
+				break;
+			}
+
+			foreach ( $posts as $post ) {
+				$current_item = $this->video_post_to_sitemap_item( $post );
+
+				if ( true === $buffer->try_to_add_item( $current_item['xml'] ) ) {
+					$last_post_id = $post->ID;
+					$buffer->view_time( $current_item['last_modified'] );
+				} else {
+					break;
+				}
+			}
+		}
+
+		// If no items were added, return false.
+		if ( true === $buffer->is_empty() ) {
+			return false;
+		}
+
+		if ( false === $buffer->is_empty() ) {
+			$this->librarian->store_sitemap_data(
+				$number,
+				JP_VIDEO_SITEMAP_TYPE,
+				$buffer->contents(),
+				$buffer->last_modified()
+			);
+		}
+
+		/*
+		 * Now report back with the ID of the last post to be
+		 * successfully added and whether there are any posts left.
+		 */
+		return array(
+			'last_id'       => $last_post_id,
+			'any_left'      => $any_posts_left,
+			'last_modified' => $buffer->last_modified(),
+		);
+	}
+
+	/**
+	 * Build and store a single page sitemap index. Return false if no index is built.
+	 *
+	 * Side effect: Create/update a sitemap index row.
+	 *
+	 * @access private
+	 * @since 4.7.0
+	 *
+	 * @param int    $number     The number of the current sitemap index.
+	 * @param int    $from_id    The greatest lower bound of the IDs of the sitemaps to be included.
+	 * @param string $timestamp  Timestamp of previous sitemap in 'YYYY-MM-DD hh:mm:ss' format.
```

Comment:
Can we call this `$datetime` instead of `$timestamp`? Timestamp to me implies an `int` (unix time).